packages/edit-site/src/components/header-edit-mode/index.js

```diff
@@ -71,12 +72,14 @@ export default function HeaderEditMode() {
 				select( editSiteStore )
 			).getEditorCanvasContainerView(),
 			isDistractionFree: getPreference( 'core', 'distractionFree' ),
+			isFixedToolbar: getPreference( 'core', 'fixedToolbar' ),
 		};
 	}, [] );
 
 	const isLargeViewport = useViewportMatch( 'medium' );
-	const { showFixedToolbar } = useShowBlockTools();
-	const showTopToolbar = isLargeViewport && showFixedToolbar;
+	const hasFixedToolbar = useHasBlockToolbar() && isFixedToolbar;
```

Comment:
Similar to @getdave suggestion. Let's assign hook values to variables and then use those in conditions.